titus-common/src/main/java/com/netflix/titus/common/network/client/internal/RetryableRestClient.java

```diff
@@ -45,20 +51,27 @@
     private final Scheduler scheduler;
     private final long timeoutMs;
     private final long retryDelayMs;
+    private final Set<HttpResponseStatus> noRetryStatuses;
 
-    public RetryableRestClient(RxRestClient delegate, int retryCount, long timeout, long retryDelay, TimeUnit timeUnit,
+    public RetryableRestClient(RxRestClient delegate, int retryCount, long timeout, long retryDelay, TimeUnit timeUnit, Set<HttpResponseStatus> noRetryStatuses,
                                RxClientMetric rxClientMetric, Scheduler scheduler) {
         this.delegate = delegate;
         this.retryCount = retryCount;
         this.rxClientMetric = rxClientMetric;
         this.scheduler = scheduler;
         this.timeoutMs = timeUnit.toMillis(timeout);
         this.retryDelayMs = timeUnit.toMillis(retryDelay);
+        this.noRetryStatuses = noRetryStatuses;
     }
 
     @Override
-    public <T> Observable<T> doGET(String relativeURI, TypeProvider<T> typeProvider) {
-        return runWithRetry(HttpMethod.GET, relativeURI, delegate.doGET(relativeURI, typeProvider));
+    public <T> Observable<T> doGET(String relativeURI, TypeProvider<T> type) {
+        return doGET(relativeURI, Collections.unmodifiableMap(new HashMap<>()), type);
```

Comment:
`Collections.unmodifiableMap(new HashMap<>())` -> `Collections.emptyMap()`- immutable empty map that is a constant such that you don't need to allocate.

Reply:
Good idea, will do.